src/server/middleware/twitter.middleware.ts

```diff
@@ -304,6 +313,172 @@ function loadTwitterData(
+function packV2Tweet(tweet: V2TweetObject, includes: V2Includes): TweetObject {
+  const entities = packEntities(tweet, includes);
+
+  const tweetObject: TweetObject = {
+    created_at: tweet.created_at,
+    id_str: tweet.id,
+    text: tweet.text,
+    // The Enterprise API entities field is not always complete, but the v2
+    // entities field is.
+    entities: entities,
+    extended_entities: entities,
+    // `display_text_range` omitted because v2 does not truncate the text.
+    favorite_count: tweet.public_metrics.like_count,
+    // `favorited` omitted because it is not available in v2..
+    in_reply_to_status_id: tweet.entities?.referenced_tweets?.find(
+      (tweet) => tweet.type === 'replied_to'
+    )?.id,
+    lang: tweet.lang,
+    reply_count: tweet.public_metrics.reply_count,
+    retweet_count: tweet.public_metrics.retweet_count,
+    source: tweet.source,
+    truncated: tweet.text.length > 140,
+    user: getUser(tweet.author_id, includes),
+  };
+
+  if (tweetObject.truncated) {
+    // Enteprise populates the extended_tweet field if the tweet is truncated,
+    // so we add it manually here for consistency.
+    tweetObject.extended_tweet = {
+      full_text: tweetObject.text,
+      display_text_range: [0, 0], // Indices not available in v2.
+      entities,
+    };
+  }
+
+  return tweetObject;
+}
+
+function packEntities(
+  tweet: V2TweetObject,
+  includes: V2Includes
+): TweetEntities {
+  const entities: TweetEntities = {};
+
+  if (tweet.entities && tweet.entities.hashtags) {
+    entities.hashtags = tweet.entities.hashtags.map(
+      (hashtag) =>
+        <TweetHashtag>{
+          indices: [hashtag.start, hashtag.end],
+          text: hashtag.tag,
+        }
+    );
+  }
+
+  if (tweet.entities && tweet.entities.urls) {
+    entities.urls = tweet.entities.urls.map(
+      (url) =>
+        <TweetUrl>{
+          display_url: url.display_url,
+          expanded_url: url.extended_url,
+          indices: [url.start, url.end],
+        }
+    );
+  }
+
+  if (tweet.entities && tweet.entities.mentions) {
+    entities.user_mentions = tweet.entities.mentions.map(
+      (mention) =>
+        <TweetUserMention>{
+          id_str: mention.id,
+          indices: [mention.start, mention.end],
+          screen_name: mention.username,
+        }
+    );
+  }
+
+  if (tweet.attachments && tweet.attachments.media_keys) {
+    entities.media = tweet.attachments.media_keys
+      .flatMap((media_key) => {
+        // v2 includes the media fields (like media_url) in a separate
+        // `includes` object, so we search there for the media item in question.
+        const media = includes.media?.find(
+          (media) => media.media_key === media_key
+        );
+        // This shouldn't happen, but we throw an error if it does.
+        if (!media) {
+          throw new Error('Unable to find media');
+        }
+        return <TweetMedia>{
+          indices: [0, 0], // Indices not available in v2.
```

Comment:
Should we remove this field if we're not using it, or keep for documentation? Alternatively we could make it optional in the TweetMedia type.

Reply:
Ah okay, thanks for the clarification. I think the bug you filed covers this issue, so I'll resolve this here.